net/sched/sch_gkprioq.c

```diff
@@ -0,0 +1,395 @@
+ /*
+ * net/sched/sch_gkprioq.c  Gatekeeper Priority Queue.
+ *
+ *		This program is free software; you can redistribute it and/or
+ *		modify it under the terms of the GNU General Public License
+ *		as published by the Free Software Foundation; either version
+ *		2 of the License, or (at your option) any later version.
+ *
+ * Authors:	Nishanth Devarajan, <ndev_2021@gmail.com>
+ *	        original idea by Michel Machado, Cody Doucette and Meng Xiang
+ *
+ */
+
+#include <linux/string.h>
+#include <linux/module.h>
+#include <linux/slab.h>
+#include <linux/types.h>
+#include <linux/kernel.h>
+#include <linux/errno.h>
+#include <linux/skbuff.h>
+#include <net/pkt_sched.h>
+#include <net/sch_generic.h>
+#include <net/dsfield.h>
+#include <net/inet_ecn.h>
+#include <asm/byteorder.h>
+
+/* Priority markers */
+#define MAX_PRIORITY 63
```

Comment:
In all the occasions `MAX_PRIORITY` is used, it is used as `MAX_PRIORITY + 1`. Just define `MAX_PRIORITY` as 64 and drop all the `+ 1` in the code.